keps/sig-scheduling/4671-gang-scheduling/README.md

```diff
@@ -0,0 +1,983 @@
+a single unit from scheduling perspective. With that, the proposed placement being a result of
+the scheduling decision of the `Workload` phase will become a `Reservation`. This will become the
+coordination point and a mechanism for multiple schedulers to share the underlying infrastructure
+addressing the requirement (4). This will also be a critical building block for workload-level
+preemption and addressing requirement (6). Finally, this will allow to address the few remaining
+corner cases around unnecessary preemption - requirement (5), such as blocking DRA resources
+(which we can't solve with NominatedNodeName). Further extensions to `Reservation` with different
+states (e.g. not yet block resources) will help with improving the scheduling accuracy.
+Finally making the binding process aware of gangs will allow to make sure the process is either
+successful or triggers workload rescheduling satisfying requirement (7).
+
+The workload-aware preemption is tightly coupled, but separate feature that will also be designed
+in a dedicated KEP. The current vision includes introducing a dedicated preemption policy (that
+will result in pods no longer being treated individually for preemption purposes) which makes it
+an additive feature. However, having a next level of details is required to ensure that we really
+have a feasible backward-compatible plan before promoting this feature to Beta.
+
+Addressing requirement (8) is the biggest effort as it requires much closer integration between
+scheduler and autoscaling components. So in the initial steps we will only focus on mitigating
+this problem with existing mechanisms (e.g. reserving resources via NominatedNodeName).
+
+However, approval for this KEP is NOT an approval for this vision. We only sketch it to show that
+we see a viable path forward from the proposed design that will not require significant rework.
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+N/A
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/scheduler`: `2025-10-02` - 81.7%
+- `k8s.io/kubernetes/pkg/scheduler/backend/queue`: `2025-10-02` - 91.4%
+- `k8s.io/kubernetes/pkg/scheduler/framework`: `2025-10-02` - 81.7%
+- `k8s.io/kubernetes/pkg/scheduler/framework/preemption`: `2025-10-02` - 64.2%
+- `k8s.io/kubernetes/pkg/scheduler/framework/util/assumecache`: `2025-10-02` - 86.2%
```

Comment:
You'll probably want to include also:
```
- k8s.io/kubernetes/pkg/apis/scheduling/v1alpha1
- k8s.io/kubernetes/pkg/apis/scheduling/validation
```
since that's where `Workload` resource will live.